graph/db/graph.go

```diff
@@ -113,6 +144,57 @@ func (c *ChannelGraph) Stop() error {
 	return nil
 }
 
+// handleTopologySubscriptions ensures that topology client subscriptions,
+// subscription cancellations and topology notifications are handled
+// synchronously.
+//
+// NOTE: this MUST be run in a goroutine.
+func (c *ChannelGraph) handleTopologySubscriptions() {
+	defer c.wg.Done()
+
+	for {
+		select {
+		// A new fully validated topology update has just arrived.
+		// We'll notify any registered clients.
+		case update := <-c.topologyUpdate:
+			c.wg.Add(1)
+			go c.handleTopologyUpdate(update)
```

Comment:
one day we will get there...but goroutine inside goroutine is bad...per-exisiting tho.

Reply:
Discussed this offline, could you please also add a TODO so we can clean this up?

Reply:
yeah def 🫡